### PR TITLE
GC AUTH YUSA: Add virtual_y role if user is active but not mapped

### PR DESCRIPTION
**Pull request by froboy**
**Related Issue/Ticket:**

No issue, discussed in slack.

> If you didn’t set up any segmentation in the backend, then it would operate with the original, simple logic [using the `virtual_y` user].

## Steps to test:

- [ ] Configure YUSA auth provider
- [ ] Do not set any mapping
- [ ] Test logging in as an active member
- [ ] Observe you are able to see VY content.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [x] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [x] This change does not contain front-end fixes.
- [x] I have flagged this PR "Needs Review" or pinged the VY devs/QA team in Slack

### CODE REVIEW

**Review by anpolimus — DISMISSED** [review]

**Review by hamrant — DISMISSED** [review]

**Comment by NadyaRudenko** [comment]
retest

**Comment by NadyaRudenko** [comment]
@froboy, @anpolimus I'm seeing a white blank screen after I try to log in. please see the video here https://monosnap.com/file/NC4kd0NixqLtuMBv4rbyCEaDPZmuGe

**Comment by froboy** [comment]
@hamrant @anpolimus well... I was going to try this, but apparently fjbot won't let me. I tried for a long time to get dependency injection for this call working on my local and for some reason I could not get it working. Could you take a look and see if you could help? I tried using similar setups from the other auth modules but it never worked.

**Comment by hamrant** [comment]
@froboy you should add `logger.factory` to `openy_gc_auth_yusa_client` service
https://github.com/fivejars/openy_gated_content/blob/2d884801b40be2baca61962c0d5c78ba09401428/modules/openy_gc_auth/modules/openy_gc_auth_yusa/openy_gc_auth_yusa.services.yml

```yml
  openy_gc_auth_yusa_client:
    class: '\Drupal\openy_gc_auth_yusa\YUSAClientService'
    arguments: ['@config.factory', '@http_client', '@logger.factory']
```

After this you need to add this param to class https://github.com/fivejars/openy_gated_content/blob/2d884801b40be2baca61962c0d5c78ba09401428/modules/openy_gc_auth/modules/openy_gc_auth_yusa/src/YUSAClientService.php

```php

  /**
   * A logger instance.
   *
   * @var \Psr\Log\LoggerInterface
   */
  protected $logger;

  /**
   * YUSAClientService constructor.
   *
   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
   *   ConfigFactoryInterface instance.
   * @param \GuzzleHttp\Client $client
   *   Guzzle client.
   * @param \Drupal\Core\Logger\LoggerChannelFactory $loggerChannelFactory
   *   Logger factory.
   */
  public function __construct(ConfigFactoryInterface $configFactory, Client $client, LoggerChannelFactory $loggerChannelFactory) {
    $this->configFactory = $configFactory;
    $this->client = $client;
    $this->logger = $loggerChannelFactory->get('openy_gc_auth_yusa');
  }
```

File: modules/openy_gc_auth/modules/openy_gc_auth_yusa/src/Form/VirtualYUSALoginForm.php [hunk 1]
```diff
@@ -247,7 +247,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       }
     }
     else {
-      if ($result['Status'] == 'Inactive') {
+      if (isset($result['Status']) && $result['Status'] == 'Inactive') {
```
Comment (froboy) [inline modules/openy_gc_auth/modules/openy_gc_auth_yusa/src/Form/VirtualYUSALoginForm.php:250]: @hamrant @anpolimus I added a check here, as there was an issue with the server while I was testing this and it was throwing bad errors. This should resolve the issue if the YUSA provider is unreachable.

**Comment by froboy** [comment]
@hamrant thank you very much! I missed adding the argument in the service before, and it should be all fixed up now.

**Review by hamrant — APPROVED** [review]

**Comment by anpolimus** [comment]
retest

**Comment by AnastasiiaPys** [comment]
retest

**Comment by NadyaRudenko** [comment]
retest

**Comment by NadyaRudenko** [comment]
retest

**Comment by NadyaRudenko** [comment]
retest

**Comment by NadyaRudenko** [comment]
@froboy, @anpolimus, @hamrant I'm not able to log in, seeing an error 
image

**Comment by NadyaRudenko** [comment]
retested, all good, thank you 🙂

**Comment by NadyaRudenko** [comment]
all good, I think, we can merge this one 🙂

**Pull request merged**